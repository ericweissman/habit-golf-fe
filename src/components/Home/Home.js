import React, { Component } from 'react';
import Feedback from '../../containers/Feedback/Feedback'

export class Home extends Component {
  state = {
    currClub: 'wedges',
  }

  changeClubs = (e) => {
    const { value } = e.target
    this.props.changeClub(value)
  }

  clubButtons = () => {
    const { club } = this.props
    const clubs = ['wedges', 'irons', 'woods']
    return clubs.map(clubType => {
      return <button className={club === clubType ? "club-btn-active" : "club-btn"} value={clubType} key={clubType} onClick={this.changeClubs}>{clubType}</button>
    })
  }

  render() {
    const { changeFilter } = this.props
    return (
      <div className="home">
        <div className="club-btns">
          {
            this.clubButtons()
          }
        </div>
        <Feedback changeFilter={changeFilter}/>
        <button onClick={() => changeFilter('ratings')} className="take-shot-btn">track shot</button>
        <button onClick={() => changeFilter('stats')} className="view-stats">view stats</button>
      </div>
    );
  }
}

export default Home;
