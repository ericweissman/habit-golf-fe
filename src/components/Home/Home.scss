.home {
  display: flex;
  flex-direction: column;
  padding-top: 35px;
  align-items: center;
}

.take-shot-btn {
  width: 165px;
  height: 60px;
  background-color: $masters-green;
  color: $masters-yellow;
  font-size: 20px;
  font-weight: 200;
  padding: 0 10px;
  border: none;
  border-radius: 50px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.club-btns {
  height: 40px;
  background-color: $masters-green;
  width: 300px;
  display: flex;
  justify-content: space-around;
  border-radius: 10px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  button {
    font-size: 18px;
    font-weight: 200;
    color: white;
  }
  .club-btn-active {
    color: $masters-yellow;
    font-weight: 700;
  }
}

.view-stats {
  @extend .take-shot-btn;
  width: 115px;
  font-size: 16px;
  height: 40px;
  margin-top: 20px;
}