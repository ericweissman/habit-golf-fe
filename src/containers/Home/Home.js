import React, { Component } from 'react';
import Ratings from '../Ratings/Ratings'
import swing from '../../images/swing.svg'
import { Query } from 'react-apollo'
import gql from "graphql-tag";
import Feedback from '../Feedback/Feedback'


const GET_PLAYERS = gql`
  {
    players {
      id
      name
    }
  }
`;


class Home extends Component {
  state = {
    currClub: 'wedges',
  }

  changeClubs = (e) => {
    const { value } = e.target
    this.props.changeClub(value)
  }



  clubButtons = () => {
    const { club } = this.props
    const clubs = ['wedges', 'irons', 'woods']
    return clubs.map(clubType => {
      return <button className={club === clubType ? "club-btn-active" : "club-btn"} value={clubType} key={clubType} onClick={this.changeClubs}>{clubType}</button>
    })
  }

  render() {
    const {changeFilter} = this.props
    return (
      // <Query query={GET_PLAYERS}>
      //   {({ loading, error, data }) => {
      //     if (loading) return "Loading...";
      //     if (error) return `Error! ${error.message}`;
      //     return (
      //       <select name="player">
      //         {data.players.map(player => (
      //           <option key={player.id} value={player.name}>
      //             {player.name}
      //           </option>
      //         ))}
      //       </select>
      //     );
      //   }}
      // </Query>

      <div className="home">
        <div className="club-btns">
          {
            this.clubButtons()
          }
        </div>
        <Feedback/>
        <button onClick={() => changeFilter('ratings')} className="take-shot-btn">Press After Your Shot</button>
      </div>
    );
  }
}

export default Home;
