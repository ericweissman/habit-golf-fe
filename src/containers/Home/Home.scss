.home {
  display: flex;
  flex-direction: column;
  padding-top: 35px;
  align-items: center;
}

.take-shot-btn {
  width: 240px;
  height: 60px;
  background-color: #fff;
  font-size: 20px;
  font-weight: 200;
  padding: 0 10px;
  border: none;
  border-radius: 50px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.club-btns {
  height: 40px;
  background-color: #fff;
  width: 300px;
  display: flex;
  justify-content: space-around;
  border-radius: 10px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  button {
    font-size: 18px;
    font-weight: 200;
  }
  .club-btn-active {
    color: $green-accent;
    font-weight: 700;
  }
}

