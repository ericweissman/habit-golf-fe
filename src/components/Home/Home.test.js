import React from 'react';
import Home from './Home';
import { shallow } from 'enzyme'

describe('Home', () => {
  let wrapper
  let mockProps = {
    changeClub: jest.fn(),
    club: 'wedges',
    changeFilter: jest.fn()
  }
  beforeEach(() => {
    wrapper = shallow(<Home {...mockProps} />)
  })

  it('should correctly match the snapshot', () => {
    wrapper = shallow(<Home {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have the correct default state', () => {
    const expectedState = { currClub: 'wedges'}
    expect(wrapper.state()).toEqual(expectedState)
  })

  it.skip('should call changeClub when changeClubs is called', () => {
    const mockEvent = {target: {
      value: 'irons'
    }}
    wrapper.instance().changeClubs(mockEvent)
    expect(wrapper.props().changeClub).toHaveBeenCalled()
  })
  
  it.skip('should call changeFilter once the track or view stats buttons are clicked', () => {
    wrapper.find('.take-shot-btn').simulate('click')
    expect(wrapper.props().changeFilter).toHaveBeenCalled()
  })
})
