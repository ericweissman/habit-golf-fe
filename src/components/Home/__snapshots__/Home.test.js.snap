// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should correctly match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="club-btns"
        >
          <button
            className="club-btn"
            onClick={[Function]}
            value="wedges"
          >
            wedges
          </button>
          <button
            className="club-btn"
            onClick={[Function]}
            value="irons"
          >
            irons
          </button>
          <button
            className="club-btn"
            onClick={[Function]}
            value="woods"
          >
            woods
          </button>
        </div>,
        <Connect(Feedback) />,
        <button
          className="take-shot-btn"
          onClick={[Function]}
        >
          track shot
        </button>,
        <button
          className="view-stats"
          onClick={[Function]}
        >
          view stats
        </button>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="club-btn"
              onClick={[Function]}
              value="wedges"
            >
              wedges
            </button>,
            <button
              className="club-btn"
              onClick={[Function]}
              value="irons"
            >
              irons
            </button>,
            <button
              className="club-btn"
              onClick={[Function]}
              value="woods"
            >
              woods
            </button>,
          ],
          "className": "club-btns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "wedges",
            "nodeType": "host",
            "props": Object {
              "children": "wedges",
              "className": "club-btn",
              "onClick": [Function],
              "value": "wedges",
            },
            "ref": null,
            "rendered": "wedges",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "irons",
            "nodeType": "host",
            "props": Object {
              "children": "irons",
              "className": "club-btn",
              "onClick": [Function],
              "value": "irons",
            },
            "ref": null,
            "rendered": "irons",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "woods",
            "nodeType": "host",
            "props": Object {
              "children": "woods",
              "className": "club-btn",
              "onClick": [Function],
              "value": "woods",
            },
            "ref": null,
            "rendered": "woods",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "track shot",
          "className": "take-shot-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "track shot",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "view stats",
          "className": "view-stats",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "view stats",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="club-btns"
          >
            <button
              className="club-btn"
              onClick={[Function]}
              value="wedges"
            >
              wedges
            </button>
            <button
              className="club-btn"
              onClick={[Function]}
              value="irons"
            >
              irons
            </button>
            <button
              className="club-btn"
              onClick={[Function]}
              value="woods"
            >
              woods
            </button>
          </div>,
          <Connect(Feedback) />,
          <button
            className="take-shot-btn"
            onClick={[Function]}
          >
            track shot
          </button>,
          <button
            className="view-stats"
            onClick={[Function]}
          >
            view stats
          </button>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="club-btn"
                onClick={[Function]}
                value="wedges"
              >
                wedges
              </button>,
              <button
                className="club-btn"
                onClick={[Function]}
                value="irons"
              >
                irons
              </button>,
              <button
                className="club-btn"
                onClick={[Function]}
                value="woods"
              >
                woods
              </button>,
            ],
            "className": "club-btns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "wedges",
              "nodeType": "host",
              "props": Object {
                "children": "wedges",
                "className": "club-btn",
                "onClick": [Function],
                "value": "wedges",
              },
              "ref": null,
              "rendered": "wedges",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "irons",
              "nodeType": "host",
              "props": Object {
                "children": "irons",
                "className": "club-btn",
                "onClick": [Function],
                "value": "irons",
              },
              "ref": null,
              "rendered": "irons",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "woods",
              "nodeType": "host",
              "props": Object {
                "children": "woods",
                "className": "club-btn",
                "onClick": [Function],
                "value": "woods",
              },
              "ref": null,
              "rendered": "woods",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "track shot",
            "className": "take-shot-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "track shot",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "view stats",
            "className": "view-stats",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "view stats",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
